src/OracleService/OracleService.cs

```diff
@@ -172,18 +181,17 @@ public JObject SubmitOracleResponse(JArray _params)
 
             if (finishedCache.ContainsKey(requestId)) throw new RpcException(-100, "Request has already finished");
 
-            using (var snapshot = Blockchain.Singleton.GetSnapshot())
-            {
-                uint height = NativeContract.Ledger.CurrentIndex(snapshot) + 1;
-                var oracles = NativeContract.RoleManagement.GetDesignatedByRole(snapshot, Role.Oracle, height);
-                if (!oracles.Any(p => p.Equals(oraclePub))) throw new RpcException(-100, $"{oraclePub} isn't an oracle node");
-                if (NativeContract.Oracle.GetRequest(snapshot, requestId) is null)
-                    throw new RpcException(-100, "Request is not found");
-                var data = Neo.Helper.Concat(oraclePub.ToArray(), BitConverter.GetBytes(requestId), txSign);
-                if (!Crypto.VerifySignature(data, msgSign, oraclePub)) throw new RpcException(-100, "Invalid sign");
-
-                AddResponseTxSign(snapshot, requestId, oraclePub, txSign);
-            }
+            var snapshot = System.StoreView;
```

Comment:
If you need to maintain the consistency of multiple reads, then you should use `System.GetSnapshot()`.

Reply:
Done

Reply:
So should we modify `RemoteNode.ProtocolHandler` https://github.com/neo-project/neo/blob/master/src/neo/Network/P2P/RemoteNode.ProtocolHandler.cs#L174 as well?